src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -478,6 +515,22 @@ private IDataView GetShuffledData(string path)
                         }
                     }
 
+                    //Process last incomplete batch
+                    if(batchIndex > 0)
+                    {
+                        var outputTensors = validationEvalRunner
+                                .AddInput(new Tensor(featureBatchPtr, featureTensorShape, TF_DataType.TF_FLOAT, featureBatchSizeInBytes), 0)
+                                .AddInput(new Tensor(labelBatchPtr, labelTensorShape, TF_DataType.TF_INT64, labelBatchSizeInBytes), 1)
+                                .Run();
+
+                        outputTensors[0].ToScalar<float>(ref accuracy);
+                        metrics.Train.Accuracy += accuracy;
+                        metrics.Train.BatchProcessedCount += 1;
+                        batchIndex = 0;
+
+                        outputTensors[0].Dispose();
+                    }
```

Comment:
At this point we know the batch size is not equal to batchSize so we will need to change the shape(featureTensorShape,labelTensorShape) of the input tensor so TF does not process the entire batch and same for batchSizeInBytes.